cmd/agent/app/server.go

```diff
@@ -39,11 +42,19 @@ func (a *Agent) run(o *options.GrpcProxyAgentOptions) error {
 		return fmt.Errorf("failed to validate agent options with %v", err)
 	}
 
-	stopCh := make(chan struct{})
-	if err := a.runProxyConnection(o, stopCh); err != nil {
+	ctx := context.Background()
+	var err error
+	var cs *agent.ClientSet
+	if cs, err = a.runProxyConnection(ctx, o); err != nil {
```

Comment:
Wondering if it would make sense to move this into app ala our server code.

Reply:
Rather than proxy connection, cluster egress might make more sense?

Reply:
I have no strong opinion on this, I can move it if you prefer.